packages/query/src/stacking/stacking.query.ts

```diff
@@ -0,0 +1,140 @@
+import { ClientParam } from '@stacks/common';
+import { StacksNetwork } from '@stacks/network';
+import { StackingClient } from '@stacks/stacking';
+import { fetchCallReadOnlyFunction, noneCV, principalCV } from '@stacks/transactions';
+import { useQuery } from '@tanstack/react-query';
+
+import { StackingQueryPrefixes } from '../query-prefixes';
+
+const defaultQueryOptions = {
+  refetchOnMount: true,
+  refetchOnReconnect: false,
+  refetchOnWindowFocus: false,
+} as const;
+
+interface StackingParams {
+  client: StackingClient;
+}
+
+interface CreateGetTransactionByIdQueryOptionsArgs {
+  senderAddress: string | null;
+  network: StacksNetwork;
+  contractAddress: string;
+  contractName: string;
+  callingContract: string;
+}
+
+export function createGetAllowanceContractCallersQueryOptions({
+  senderAddress,
+  network,
+  contractAddress,
+  contractName,
+  callingContract,
+  client,
+}: CreateGetTransactionByIdQueryOptionsArgs & ClientParam) {
+  return {
+    queryKey: [
+      StackingQueryPrefixes.GetAllowanceContractCallers,
+      senderAddress,
+      callingContract,
+      network,
+      contractAddress,
+      contractName,
+    ],
+    queryFn: () => {
+      if (senderAddress) {
+        return fetchCallReadOnlyFunction({
+          contractAddress,
+          contractName,
+          functionName: 'get-allowance-contract-callers',
+          functionArgs: [principalCV(senderAddress), principalCV(callingContract)],
+          senderAddress,
+          network,
+          client,
+        });
+      } else {
+        return Promise.resolve(noneCV());
+      }
+    },
+    ...defaultQueryOptions,
+  } as const;
```

Comment:
could we use `satisfies `UseQueryOptions` here for stronger typing?

`refetchOnRefreshy` is a valid prop here and won't cause type err